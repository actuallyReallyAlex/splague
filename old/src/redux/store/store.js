import { createStore, combineReducers } from 'redux'
import cureReducer from '../reducers/cure'
import logReducer from '../reducers/log'
import plagueReducer from '../reducers/plague'
import playerReducer from '../reducers/player'
import uiReducer from '../reducers/ui'
import worldReducer from '../reducers/world'

const store = createStore(
  combineReducers({
    cure: cureReducer,
    log: logReducer,
    plague: plagueReducer,
    player: playerReducer,
    ui: uiReducer,
    world: worldReducer
  }),
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)

export default store
