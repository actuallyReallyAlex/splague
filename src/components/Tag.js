import React from 'react'
import PropTypes from 'prop-types'
import { Box, Text } from 'grommet'

export const Tag = ({ label }) => {
  return (
    <Box
      background="dark-6"
      pad={{ horizontal: '6px', vertical: '2px' }}
      round="3px"
    >
      <Text
        color="white"
        size="xsmall"
        style={{ letterSpacing: '-1px' }}
        weight="bold"
      >
        {label}
      </Text>
    </Box>
  )
}

Tag.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
}

export default Tag
