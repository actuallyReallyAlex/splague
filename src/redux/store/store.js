import { createStore, combineReducers } from 'redux'
import WorldReducer from '../reducers/world'

const store = createStore(
  combineReducers({
    world: WorldReducer
  }),
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)

export default store
