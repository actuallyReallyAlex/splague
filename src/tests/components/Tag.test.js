import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import store from '../../redux/store/store'
import Tag from '../../components/Tag'

describe('<Tag />', () => {
  it('Should render snapshot of Tag component.', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <Tag label="100" />
        </Provider>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
