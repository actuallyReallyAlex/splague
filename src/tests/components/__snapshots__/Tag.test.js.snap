// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> Should render snapshot of Tag component. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  background: #999999;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 6px;
  padding-right: 6px;
  padding-top: 2px;
  padding-bottom: 2px;
  border-radius: 3px;
}

.c1 {
  font-size: 12px;
  line-height: 18px;
  color: #FFFFFF;
  font-weight: bold;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-left: 6px;
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-top: 2px;
    padding-bottom: 2px;
  }
}

<div
  className="c0"
>
  <span
    className="c1"
    size="xsmall"
    style={
      Object {
        "letterSpacing": "-1px",
      }
    }
  >
    100
  </span>
</div>
`;
